src/main/java/org/openstreetmap/atlas/checks/base/BaseCheck.java

```diff
@@ -364,6 +367,37 @@ protected CheckFlag createFlag(final AtlasObject object, final List<String> inst
                 instructions, points);
     }
 
+    /**
+     * If this object is a way-sectioned edge, return a set of all edges that originated from the
+     * set. Otherwise, return a singleton containing this edge.
```

Comment:
This could return a singleton containing an AtlasObject other than an edge-- maybe we could reword the last sentence to be "return a singleton containing this AtlasObject." Which also makes me wonder. Since this method is only practical for Edges, why isn't the parameter of type Edge?

Reply:
Nice catch on the typo there, thanks!

Your other question is a good one. For the other function `gatherWaySectionedEdges(final Set<? extends AtlasObject> object)`, it's clear that we want to handle both edges and non-edges inside the set. I could imagine a check that flags an edge and a couple of points, for example. Instead of forcing the developer of the check to filter through and pull out all of the edges, they can just pass in the set and we handle all of the filtering here. Given that design decision, it 
 - a) was the most helpful in writing the `Set<? extends AtlasObject>` function 
 - b) seemed to be nicely symmetrical with the `Set<? extends AtlasObject>` pattern
I think all the lines of code need to be here in one form or another, but I'm not 100% sold on their current location. A couple other options I see would be to:
 - remove this function entirely (migrating the code into the more important `Set` function)
 - make this function only take in `Edge`s, (migrating the instanceof check and singleton call to the `Set` function).
 - something else that I'm totally missing 😆 

What do you think?

Reply:
From a development standpoint, I'm just not sure I'd expect gatherWaySectionedEdges() to return anything if I passing in something other than an Edge as an argument. I'm thinking that if we move the instanceof check and the singleton call to the Set method and evaluate it using ternary operator then it might be cleaner/more modular.

If we do this I don't think we'll need a function to handle a single Atlas Object or Edge-- if a user wants the way sectioned edges of a single Edge, they can just stick to using the OsmWayWalker directly.

Reply:
I could be clearer- here's roughly what I was suggesting:
```
objects.stream().flatMap(object -> object instanceof Edge ? Collections.unmodifiableSet(new OsmWayWalker((Edge) object).collectEdges()).stream() : Collections.singleton(object).stream())
	                .collect(Collectors.toSet());
```
with no need now for the function on line 378.

Reply:
Yep, got it. 

I think that `createFlag(new OsmWayWalker((Edge) object).collectEdges())` is a fine usage pattern and I'm happy with that as the end result for a check developer. At the moment, they'll need to do `createFlag(Collections.unmodifiableSet(new OsmWayWalker((Edge) object).collectEdges())` which I don't like very much. The proper solution for this would be to make `createFlag` accept `Set<? extends AtlasObject>` as mentioned above. 

Maybe that should be the first step here -- it seems like it'll be much cleaner then.

Reply:
Yeah I agree. Definitely up for more discussion on that in a (hopefully) future PR!

Reply:
Resolving this since neither helper function ended up being super necessary :)